scripts/dts/edtlib.py

```diff
@@ -755,6 +759,11 @@ def name(self):
         "See the class docstring"
         return self._node.name
 
+    @property
+    def is_device(self):
+        "See the class docstring"
+        return self._binding and self._binding.get('is-device', False)
```

Comment:
Maybe this should default to True. As long as we have a matching-compat, we're likely to be a struct device.

Reply:
Defaulting to True will break  #22255. #22255 introduces default bindings for special nodes. These special nodes would be recognized as devices, but are none.

Reply:
Noted.